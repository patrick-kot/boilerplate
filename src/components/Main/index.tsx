import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de um átomo e React Avançado escrito ao lado"
    />
    <S.Title>React Avançado</S.Title>
    <S.Description>
      TypeScript, ReactJS, NextJS e Styled Components
    </S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Ilustração de uma pessoa sentada de costas olhando para uma tela de códigos"
    />
  </S.Wrapper>
)

export default Main
