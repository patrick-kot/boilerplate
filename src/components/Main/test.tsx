import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('deve renderizar o heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avanÃ§ado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('deve renderizar a cor corretamente', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
