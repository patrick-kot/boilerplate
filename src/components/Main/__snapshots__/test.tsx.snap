// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> deve renderizar o heading 1`] = `
<main>
  <h2>
    React Avan√ßado
  </h2>
</main>
`
